### PR TITLE
#502 Moved ci to AWS code build

### PR DESCRIPTION
**Pull request by tomuben**
fixes #502

Initial support for AWS Ci builds.

### CODE REVIEW

File: .github/workflows/check_starter_scripts.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    - name: Install Poetry
+      uses: abatilo/actions-poetry@v2.0.0
+      with:
+        poetry-version: 1.1.13
+
+    - name: Setup poetry env
+      run:  bash scripts/setup_poetry_env.sh "python3.8"
+
+ #   - name: Run check_if_starter_scripts_are_up_to_date.sh
```
Comment (tkilias) [inline .github/workflows/check_starter_scripts.yaml:28]: Why are these comments

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+keywords = ['exasol', 'udf', 'script-languages']
+
+[tool.poetry.dependencies]
+python = ">=3.8,<4"
+importlib-resources = ">=5.4.0"
+exasol_script_languages_container_tool = { url = "https://github.com/exasol/script-languages-container-tool/releases/download/0.12.0/exasol_script_languages_container_tool-0.12.0-py3-none-any.whl" }
+exasol-script-languages-container-ci = {git = "https://github.com/exasol/script-languages-container-ci.git", branch="feature/9_implement_run-release_command"}
+exasol-script-languages-container-ci-setup = {git = "https://github.com/exasol/script-languages-container-ci-setup.git", branch="feature/3_addess_release_code_build_and_command"}
+#exasol-script-languages-container-ci = { path = "/home/thomas/Work/script-languages-container-ci", develop = false}
```
Comment (tkilias) [inline pyproject.toml:25]: Cleanup

**Review by tkilias — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 3]
```diff
@@ -11,4 +11,6 @@ emulator/zmqcontainer_pb2.py
 .build_output/
 luigi.cfg
 *.iml
-exaudfclient/base/file.txt
\ No newline at end of file
+/codebuild_build.sh
+exaudfclient/base/file.txt
+dist
```
Comment (Nicoretti) [inline .gitignore:16]: missing newline?

File: aws-code-build/build_config.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{"build_ignore":
+  {"ignored_paths":
+    ["google-cloud-build", "python_client", ".gitignore", "contributing.md", ".dockerignore", "LICENSE",
+      "pyproject.toml", "README.md", "find_duplicate_error_codes.sh", "find_error_codes.sh",
+      "find_highest_error_codes_per_module.sh", "find_incomplete_error_codes.sh", "find_next_error_code_per_module.sh",
+      "show_code_arround_error_codes.sh", "visualize_task_dependencies.sh", "udf-script-signature-generator",
+      ".github", "cpp_client", "emulator", "exaslct_scripts", "githooks"]
+  }
+}
```
Comment (Nicoretti) [inline aws-code-build/build_config.json:9]: missing newline

File: aws-code-build/ci/buildspec.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      buildspec: ./aws-code-build/ci/build_buildspec.yaml
+      privileged-mode: true
+      type: BUILD_GENERAL1_MEDIUM
+    - identifier: build_python_38_minimal_EXASOL_620
+      env:
+        variables:
+          FLAVOR: python-3.8-minimal-EXASOL-6.2.0
+      buildspec: ./aws-code-build/ci/build_buildspec.yaml
+      privileged-mode: true
+      type: BUILD_GENERAL1_MEDIUM
```
Comment (Nicoretti) [inline aws-code-build/ci/buildspec.yaml:36]: missing newline

File: aws-code-build/run_local/get.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ popd
+ rm -rf aws-codebuild-docker-images
+ popd
+fi
+
+echo Ready.
+echo Now you can run the AWS code build locally with:
+echo "'./aws-code-build/run_local/run_local.sh in the root folder (where buildspec.yml exists)"
+
+popd &> /dev/null
```
Comment (Nicoretti) [inline aws-code-build/run_local/get.sh:25]: missing newline

File: aws-code-build/run_local/run_local.env [hunk 7]
```diff
@@ -0,0 +1,5 @@
+FLAVOR=python-3.8-minimal-EXASOL-6.2.0
+CODEBUILD_SOURCE_VERSION=4018a9005ed8b83ebf7bb75b7507be9316972a32
+CODEBUILD_WEBHOOK_HEAD_REF=refs/heads/refactoring/502_move_ci_to_AWS_Code_Build
+BUILD_DOCKER_REPOSITORY="exatk/script-languages-build-cache"
+RELEASE_DOCKER_REPOSITORY="exasol/script-language-container"
```
Comment (Nicoretti) [inline aws-code-build/run_local/run_local.env:5]: missing newline

File: aws-code-build/run_local/run_local.sh [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+set -euo pipefail
+
+BUILD_SPEC=$1
+AWS_PROFILE=$2
+SCRIPT_DIR="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+pushd "$SCRIPT_DIR/../../" &>/dev/null
+#Artifacts will be stored under /tmp
+./codebuild_build.sh -e ./aws-code-build/run_local/run_local.env -d -m -c -i aws/codebuild/ubuntu/standard:5.0 -b "$BUILD_SPEC" -s "." -p $AWS_PROFILE -a /tmp
+popd > /dev/null
```
Comment (Nicoretti) [inline aws-code-build/run_local/run_local.sh:12]: missing newline

File: githooks/update_exaslct_starter_scripts.sh [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+SCRIPT_DIR="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+ROOT_DIR="$SCRIPT_DIR/.."
+
+pushd "$ROOT_DIR" > /dev/null
+
+poetry run python3 -m exasol_script_languages_container_tool.main install-starter-scripts --install-path "." --force-install
+git add "./exaslct"
+git add "./exaslct_scripts"
+
+popd > /dev/null
```
Comment (Nicoretti) [inline githooks/update_exaslct_starter_scripts.sh:17]: missing newline

File: scripts/setup_poetry_env.sh [hunk 10]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+PYTHON_VERSION=$1
+
+PYTHON_BIN=$(command -v "$PYTHON_VERSION")
+poetry env use "$PYTHON_BIN"
+poetry install
```
Comment (Nicoretti) [inline scripts/setup_poetry_env.sh:10]: missing newline

File: aws-code-build/build_config.json [hunk 11]
```diff
@@ -0,0 +1,9 @@
+{"build_ignore":
+  {"ignored_paths":
+    ["google-cloud-build", "python_client", ".gitignore", "contributing.md", ".dockerignore", "LICENSE",
+      "pyproject.toml", "README.md", "find_duplicate_error_codes.sh", "find_error_codes.sh",
```
[the remark this answers is not in the document]
Comment (tomuben) [reply aws-code-build/build_config.json:4]: Done

File: aws-code-build/run_local/get.sh [hunk 12]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+set -euo pipefail
+
+pushd "../.." &>/dev/null
+
+echo Pulling codebuild skript...
```
[the remark this answers is not in the document]
Comment (tomuben) [reply aws-code-build/run_local/get.sh:6]: Done

File: pyproject.toml (hunk 2 above)
Comment (tomuben) [reply pyproject.toml:25]: Once this PR and https://github.com/exasol/script-languages-container-ci-setup/pull/5 is approved I will create releases for exasol-script-languages-container-ci and exasol-script-languages-container-ci-setup and then update the dependencies here accordingly.

File: aws-code-build/build_config.json (hunk 4 above)
Comment (tomuben) [reply aws-code-build/build_config.json:9]: Done

File: aws-code-build/ci/buildspec.yaml (hunk 5 above)
Comment (tomuben) [reply aws-code-build/ci/buildspec.yaml:36]: Needs to be fixed in the other project. I will fix it there....

File: aws-code-build/run_local/run_local.env (hunk 7 above)
Comment (tomuben) [reply aws-code-build/run_local/run_local.env:5]: Done

File: aws-code-build/run_local/run_local.sh (hunk 8 above)
Comment (tomuben) [reply aws-code-build/run_local/run_local.sh:12]: Done

File: githooks/update_exaslct_starter_scripts.sh (hunk 9 above)
Comment (tomuben) [reply githooks/update_exaslct_starter_scripts.sh:17]: Done

File: scripts/setup_poetry_env.sh (hunk 10 above)
Comment (tomuben) [reply scripts/setup_poetry_env.sh]: Fixed

File: aws-code-build/run_local/run_local.env [hunk 13]
```diff
@@ -0,0 +1,5 @@
+FLAVOR=python-3.8-minimal-EXASOL-6.2.0
```
Comment (Nicoretti) [inline aws-code-build/run_local/run_local.env:1]: is EXASOL 6.2.0 still correct?

Comment (tomuben) [reply aws-code-build/run_local/run_local.env:1]: More or less, we already had a discussion with PM and as a consequence opened https://github.com/exasol/script-languages-release/issues/566 to fix all custom flavor names.

**Review by tkilias — DISMISSED** [review]

**Review by tkilias — APPROVED** [review]

**Pull request merged**